app/uk/gov/hmrc/individualsincomeapi/play/RequestHeaderUtils.scala

```diff
@@ -43,6 +44,12 @@ object RequestHeaderUtils {
       case None => throw new BadRequestException("CorrelationId is required")
     }
 
+  def maybeCorrelationId(requestHeader: RequestHeader) =
+    Try(validateCorrelationId(requestHeader)) match {
```

Comment:
Can we be sure an exception thrown by validateCorrelationId will be handled here? This method should return None when the correlation id cannot be extracted. We do not want an exception here. If it swallows the exception with the `Try` then that is fine.